### DISCUSSION TITLE
Strelka failed to compile yara rules

### DISCUSSION THREAD

**Discussion by Tedus007** [Security-Onion-Solutions/securityonion#5498] [Unsupported Versions] [answered]
Dear Team,

I've installed new SO 2.3.7 and whenever I check the strelka logs I can see yara compile error, however I can't detect which rule causing the issue.

I tried to delete all the rules from the signature-base directory but salt is injecting the rules back.

 cat /nsm/strelka/log/strelka.log | jq .

    "yara": {
      "elapsed": 0.704228,
      "flags": [
        "compiling_error"
      ]

**Comment by dougburks** [comment, marked as answer]
This is a known issue and we are working on a fix.

**Comment by Tedus007** [comment]
Thanks Doug,

For now is there any known workaround can be done until a fix is released so I can get yara working again?

**Comment by Tedus007** [comment]
Hello, 
I've followed this link and run the script against all yara rules and no rules reporting errors 

https://github.com/Security-Onion-Solutions/securityonion/discussions/4580

All rules are valid with no syntax error .

I'm not using any custom rules.

Also I've added a file for the strelka unprocessed directory for testing and always getting same error /nsm/strelka/unprocessed

Still getting same error
'{"file":{"depth":0,"flavors":{"mime":["text/x-msdos-batch"]},"scanners":["ScanBatch","ScanEntropy","ScanHash","ScanHeader","ScanYara"],"size":1703,"tree":{"node":"3903021b-9a60-44e5-8c61-d401ecbb85ff","root":"3903021b-9a60-44e5-8c61-d401ecbb85ff"}},"request":{"attributes":{"filename":"/nsm/strelka/staging/ben10.bat"},"client":"go-filestream","id":"3903021b-9a60-44e5-8c61-d401ecbb85ff","source":"40d31cbfb1d6","time":1632042515},"scan":{"batch":{"elapsed":0.004587,"keywords":["echo","set"],"text":["off","\\n","set","gayftg","=","hffuileyn","hhmzzkd","s","kpfiyrjl","t","mpcprrpso","a","jseslg","r","bnczuzpu","drpunjot","acrfl","C","dzjdqxd",":","jgxcdk","/","fdbvnh","W","autfn","I","sxahxc","N","bdzufwp","D","xlfrxlfpy","O","vdggcf","swakiqk","S","mraew","xiunqngys","hdfin","y","rwmijpp","trrpoahn","miyrdbk","e","vucdqkm","m","oocuc","3","zxmbacdpv","2","xkdvjwr","spfqpqz","c","viecm","yvhrsqgob","l","ldmmpdn","rwnji",".","jinaygp","lqovf","x","ykxvkqaj"],"tokens":["Token.Punctuation","Token.Keyword","Token.Text","Token.Name.Variable"],"variables":["pywfverpj","%pywfverpj%","%gayftg%","%hffuileyn%","%rwmijpp%","%trrpoahn%","%viecm%","%jseslg%","%drpunjot%","%acrfl%","%dzjdqxd%","%xkdvjwr%","%vdggcf%","%autfn%","%sxahxc%","%bdzufwp%","%xlfrxlfpy%","%xiunqngys%","%hdfin%","%ykxvkqaj%","%vucdqkm%","%oocuc%","%zxmbacdpv%","%ldmmpdn%","%yvhrsqgob%","%rwnji%","%lqovf%"]},"entropy":{"elapsed":0.00006,"entropy":4.458983383670524},"hash":{"elapsed":0.003297,"md5":"32bbba8fb06346c840f5c326507688e5","sha1":"015872b23079112d79728a9c3882cf30d04e46c2","sha256":"8a04fe710ba170d4af4971e00961a809afdfe082784a51fd077c47097f2a6ccc","ssdeep":"48:Mm4qjq+qbIqoqNnqTSqvqDWqiqTquq+qO/qNvqlqB3qRqkqaqAQqVqjqAqOWq3qE:1G2X2532M2"},"header":{"elapsed":0.000044,"header":"@echo off\nset pywfverpj=set\n%pywfverpj% gayftg= \n%"},"**yara":{"elapsed":0.124329,"flags":["compiling_error"**]}}}'

**Comment by Tedus007** [comment]
The below fix is being done by Wes, I tested the files in my deployment and it works perfectly.
https://github.com/Security-Onion-Solutions/securityonion/pull/5521/
